### ISSUE TITLE
cannot unmarshal string into Go struct field ServiceCheck.Interval of type time.Duration

### ISSUE THREAD

**Issue by MatthewPhinney** [hashicorp/nomad-java-sdk#24]
I am trying to register a job I created from a JSON file:
```
final Job job = Job.fromJson(jobSpecification);
jobsApi.register(job);
```
This results in the error
```
com.hashicorp.nomad.javasdk.ErrorResponseException: PUT https://<host>/v1/jobs resulted in error response status HTTP/1.1 400 Bad Request: json: cannot unmarshal string into Go struct field ServiceCheck.Interval of type time.Duration
```
Either my job is somehow malformed, or the Java data type for `ServiceCheck.Interval` (looks like a String) doesn't work with Go's `time.Duration`. I also tested this with the example job JSON I found [here](https://www.nomadproject.io/api/json-jobs.html) and got the same error. I'm using version `0.8.6` of the SDK.

Can you please let me know how to fix this?

**Comment by cgbaker** [comment]
Will look at this immediately.

**Comment by cgbaker** [comment]
Okay, I changed something in the generator that turns the Go API into the Java SDK in order to fix JSON deserialization, but apparently it simultaneously broke JSON serialization somewhere else. I'm fixing this now, will release an 0.8.6.1 when it's done.

**Comment by cgbaker** [comment]
Okay, @MatthewPhinney, this should be corrected in 0.8.6.1, which has been released. Apologies for breaking this... let me know if there are any other problems.
